modules/bibupload/lib/bibupload.py

```diff
@@ -1212,27 +1212,33 @@ def check_record_doi_is_unique(rec_id, record):
     Return (boolean, msg) where 'boolean' would be True if the DOI is
     unique.
     """
-    record_dois = record_extract_dois(record)
-    if record_dois:
-        matching_recids = set()
-        for record_doi in record_dois:
-            possible_recid = find_record_from_doi(record_doi)
-            if possible_recid:
-                matching_recids.add(possible_recid)
-        if len(matching_recids) > 1:
-            # Oops, this record refers to DOI existing in multiple records.
-            msg = "   Failed: Multiple records found in the" \
+    try:
+        from invenio.config import CFG_BIBUPLOAD_DOI_CHECKING
```

Comment:
A try/except technique for imports is necessary in `maint-x.y` branches where we promised not to break API/DB/config.  We use techniques like this:

- https://github.com/inveniosoftware/invenio/blob/maint-1.2/modules/bibdocfile/lib/bibdocfile_config.py#L20-L31

I fully agree the import can be up in the file though.